thoth/investigator/investigate_unresolved_package.py

```diff
@@ -115,34 +117,50 @@ def parse_unresolved_package_message(unresolved_package: MessageBase) -> None:
     """Parse unresolved package message."""
     package_name = unresolved_package.package_name
     package_version = unresolved_package.package_version
-    indexes: Optional[List[str]] = unresolved_package.index_url
+    requested_indexes: Optional[List[str]] = unresolved_package.index_url
     solver = unresolved_package.solver
 
     graph = GraphDatabase()
     graph.connect()
 
     registered_indexes: List[str] = graph.get_python_package_index_urls_all()
 
-    if indexes:
-        _LOGGER.warning(f"User requested indexes are: {indexes}")
+    indexes = registered_indexes
 
-    if not package_version:
-        packages = package_name
+    if not requested_indexes:
+        _LOGGER.info("Scheduling Thoth registered indexes...")
     else:
-        packages = f"{package_name}==={package_version}"
-
-    is_scheduled = _schedule_solver_with_priority(packages=packages, indexes=registered_indexes, solver=solver)
+        if all(index_url in registered_indexes for index_url in requested_indexes):
+            indexes = requested_indexes
+            _LOGGER.info("Using requested indexes...")
+        else:
+            indexes_not_registered = set(requested_indexes) - set(registered_indexes)
+            if indexes_not_registered:
+                _LOGGER.warning(f"User requested indexes not registered: {indexes_not_registered}")
```

Comment:
Do we want to enable this? It can be a security hole.

Reply:
:grimacing: didnt know that

Reply:
Removed :)